Documentation update

jzuniga-amd opened the pull request:
Prepares documentation to be consistent with the new APIs that will be introduced soon

jzuniga-amd committed:
update documentation

EdDAzevedo commented on library/include/rocsolver/rocsolver-functions.h:
```diff
@@ -4223,39 +4224,40 @@ ROCSOLVER_EXPORT rocblas_status rocsolver_zgetf2_npvt(rocblas_handle handle,
 
     \f[
-        A_j = L_jU_j
+        A_l = L_lU_l
```
Should it be A(l) = L(l) U(l)?       Note the lower case  "el" (l) can look like "one" (1) in some fonts.

cgmb replied:
If you click on "Details" next to docs/readthedocs.org:rocsolver in the GitHub checks for this PR, you can see how the documentation will be rendered. Here's a direct link to this function: https://rocsolver--459.org.readthedocs.build/en/459/api_lapacklike.html#c.rocsolver_zgetf2_npvt_batched

It mostly looks ok, but the varying subscript height in `L_lL'_l` bugs me. That gets rendered as $L_lL'_l$. This is because the first variable has a superscript and the second does not. In those equations, I'd suggest giving all variables in the equation an empty superscript. `L^{}_lL'_l` renders as $L^{}_lL'_l$.

EdDAzevedo commented on library/include/rocsolver/rocsolver-functions.h:
```diff
@@ -5853,54 +5856,54 @@ ROCSOLVER_EXPORT rocblas_status rocsolver_zgeql2(rocblas_handle handle,
     @param[in]
     m           rocblas_int. m >= 0.\n
-                The number of rows of all the matrices A_j in the batch.
+                The number of rows of all the matrices A_l in the batch.
```
Should these be \f$A_l\f$  with escape to latex  instead of just A_l ?  Just checking.

jzuniga-amd replied:
The problem is that, for some reason, because of the newline (\n) that we use to separate the type of the parameter from the description of the parameter, doxygen introduces newlines before the latex formulas as well.
So, it will look like this:
>rocblas_int. m>=0.
>The number of rows of all matrices
>$A_l$ in the batch.

If we remove the \n it will look like this:
>rocblas_int. m>=0. The number of rows of all matrices $A_l$ in the batch

Ideally, what I would like is this:
>rocblas_int. m>=0.
>The number of rows of all matrices $A_l$ in the batch.

but I don't know how to make it work... Any ideas?
Do you mind having everything in a single line (i.e. remove \n)?
It would be good to have math formulas in the parameter description as well because the `_l` definitely  looks ugly.

EdDAzevedo replied:
Have you tried "\linebreak" command in latex?

You can also experiment with "A_\ell" for "A_l" to see which one do you prefer.

jzuniga-amd replied:
Problem is not in latex. The problem is doxygen introducing line breaks where it shouldn't (or where I don't want them)

cgmb commented on docs/source/api_auxiliaryfunc.rst:
```diff
@@ -17,14 +17,22 @@ The auxiliary functions are divided into the following categories:
 
+    * i, j, and k are used as general purpose indices. In some legacy LAPACK APIs, k could be
+      a parameter indicating some problem/matrix dimension.
+    * Depending on the context, when it is necessary to index rows, columns and blocks or submatrices, 
+      i is assigned to rows, j to columns and k to blocks. l is always used to index 
+      matrices/problems in a batch. 
```
Would it be ridiculous to suggest using $A_b$ instead of $A_l$? I feel like with small fonts, $A_l$ may be difficult to distinguish from $A_1$ or $A_i$.

tfalders replied:
I'd also prefer a different subscript to `l` for the batch index. Using `b` makes sense to me.

jzuniga-amd replied:
Normally the first letters of the alphabet are never used as indices in math, but I don't really see a problem; I can change `_l` for `_b` if we agree... `_l` looks bad when it is not in latex. 
Now, if we can find a way to have latex formulas everywhere (see my comment above), do you still want to use `_b`? I personally don't see a problem with $A_l$ $A_1$ or $A_i$.

jzuniga-amd replied:
>You can also experiment with "A_\ell" for "A_l" to see which one do you prefer.

Yeah, I like better `A_\ell`. It is more distinguishable from `A_i` and `A_1` , I think
$A_\ell$, $A_1$, $A_i$

I would go with `A_\ell` instead of `A_b` if we need to change, but no strong preference.

tfalders replied:
Yeah, now that I think about it, using `A_b` might cause confusion in cases where `b` is the name of the right-hand solution vector. Hmmm....

cgmb replied:
I like $A_\ell$. I see that in Fortran, variables starting with the letters i, j, k, l, m, or n are of type integer by default, which I suppose is strong cultural precedent.

cgmb commented on library/include/rocsolver/rocsolver-functions.h:
```diff
@@ -4223,39 +4224,40 @@ ROCSOLVER_EXPORT rocblas_status rocsolver_zgetf2_npvt(rocblas_handle handle,
     could be executed with small and mid-size matrices if optimizations are enabled (default option). For more details, see the
     "Tuning rocSOLVER performance" section of the Library Design Guide).
```
It's beyond the original scope of your change, but this should probably be a link.

jzuniga-amd replied:
Will change this once the description of the related parameters is completed so we have the correct anchor for the link

jzuniga-amd committed:
add empty superscripts when needed

jzuniga-amd committed:
remove line breaks in parameter descriptions (test)

jzuniga-amd commented:
I added empty superscripts when needed. It looks good on my local build, but apparently it has no effect on the build in readthedocs :(

I also removed the line breaks in the parameter descriptions; it looks OK. If you don't mind we can leave it; this will allow us to introduce latex (and `_\ell`) in the parameter descriptions without the undesired line break. Have a look at the `ipiv` description in `laswp` function, for example.

Let me know what you think.

cgmb reviewed (approved):
I didn't review the whole thing after the recent updates, but I don't have any further suggestions.
